src/Compilers/Core/Portable/Symbols/INamedTypeSymbol.cs

```diff
@@ -72,6 +72,8 @@ public interface INamedTypeSymbol : ITypeSymbol
         /// </summary>
         ImmutableArray<ITypeSymbol> TypeArguments { get; }
 
+        ImmutableArray<Nullability> TypeArgumentNullability { get; }
```

Comment:
It feels like there should be an "s" somewhere in the name to make it plural.

Reply:
I agree with the feeling, but I couldn't find a spot that felt good. `TypeArgumentNullabilities` is the closes I got, thoughts?

---
In reply to: [205846588](https://github.com/dotnet/roslyn/pull/28860#discussion_r205846588) [](ancestors = 205846588)

Reply:
Is there any reason we aren't doing some struct that holds a TypeSymbol and it's nullability, and having an ImmutableArray of that struct to hold both at once? My gut right now is saying the IDE is immediately going to create such a thing, and use that wherever we're using ITypeSymbols today.

Reply:
Such a struct/interface to represent the pair (type+nullability) seems appealing, but I don't see how to introduce it without breaking existing API.